modules/webbrowser/interaction/cefkeyboardmapping.cpp

```diff
@@ -0,0 +1,151 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/webbrowser/interaction/cefkeyboardmapping.h>
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include "include/cef_base.h"
+#include <warn/pop>
+
+namespace inviwo {
+
+namespace cef {
+
+int mapKey(IvwKey key) {
+    switch (key) {
+        case IvwKey::Backspace:
+            return 8;
+        case IvwKey::Tab:
+            return 9;
+        case IvwKey::Enter:
+            return 13;
+        case IvwKey::Left:
+            return 37;
+        case IvwKey::Up:
+            return 38;
+        case IvwKey::Right:
+            return 39;
+        case IvwKey::Down:
+            return 40;
+        case IvwKey::Delete:
+            return 46;
+        case IvwKey::A:
+            return 97;
+        default:
+            return static_cast<int>(key);
+    }
+}
+
+unsigned int keyModifiers(KeyModifiers modifiers, IvwKey key) {
+    unsigned int cefModifiers = 0;
+    if (modifiers & KeyModifier::Control) {
+        cefModifiers |= EVENTFLAG_CONTROL_DOWN;
+    } else if (modifiers & KeyModifier::Shift) {
+        cefModifiers |= EVENTFLAG_SHIFT_DOWN;
+    } else if (modifiers & KeyModifier::Alt) {
+        cefModifiers |= EVENTFLAG_ALT_DOWN;
+    } else if (modifiers & KeyModifier::Super) {
+        cefModifiers |= EVENTFLAG_COMMAND_DOWN;
+    } else if (modifiers & KeyModifier::Menu) {
+        cefModifiers |= EVENTFLAG_ALT_DOWN;
+    } else if (modifiers & KeyModifier::Meta) {
+    }
+
+    if (key == IvwKey::CapsLock) {
+        cefModifiers |= EVENTFLAG_CAPS_LOCK_ON;
+    } else if (key == IvwKey::NumLock) {
+        cefModifiers |= EVENTFLAG_NUM_LOCK_ON;
+    }
+    switch (key) {
+        case IvwKey::Unknown:
+            break;
+        case IvwKey::Insert:
+        case IvwKey::Delete:
+        case IvwKey::Right:
+        case IvwKey::Left:
+        case IvwKey::Down:
+        case IvwKey::Up:
+        case IvwKey::PageUp:
+        case IvwKey::PageDown:
+        case IvwKey::Home:
+        case IvwKey::End:
+        case IvwKey::CapsLock:
+        case IvwKey::ScrollLock:
+        case IvwKey::NumLock:
+        case IvwKey::PrintScreen:
+        case IvwKey::Pause:
+            cefModifiers |= EVENTFLAG_IS_KEY_PAD;
```

Comment:
for the pause key?

Reply:
At the moment we do no distinguish between regular and numpad keys. Qt provides a key modifier for numpad keys though.